api/trace/b3_propagator.go

```diff
@@ -34,185 +34,295 @@ const (
 	b3TraceIDPadding = "0000000000000000"
 )
 
+var (
+	empty = EmptySpanContext()
+
+	errInvalidSampledByte        = errors.New("invalid B3 Sampled found")
+	errInvalidSampledHeader      = errors.New("invalid B3 Sampled header found")
+	errInvalidTraceIDHeader      = errors.New("invalid B3 TraceID header found")
+	errInvalidSpanIDHeader       = errors.New("invalid B3 SpanID header found")
+	errInvalidParentSpanIDHeader = errors.New("invalid B3 ParentSpanID header found")
+	errInvalidScope              = errors.New("require either both TraceID and SpanID or none")
+	errInvalidScopeParent        = errors.New("ParentSpanID requires both TraceID and SpanID to be available")
+	errInvalidScopeParentSingle  = errors.New("ParentSpanID requires TraceID, SpanID and Sampled to be available")
+	errEmptyContext              = errors.New("empty request context")
+	errInvalidTraceIDValue       = errors.New("invalid B3 TraceID value found")
+	errInvalidSpanIDValue        = errors.New("invalid B3 SpanID value found")
+	errInvalidParentSpanIDValue  = errors.New("invalid B3 ParentSpanID value found")
+)
+
+// B3Encoding is a bitmask representation of the B3 encoding type.
+type B3Encoding uint8
+
+const (
+	// MultipleHeader is a B3 encoding that uses multiple headers to
+	// transmit tracing information all prefixed with `x-b3-`.
+	MultipleHeader B3Encoding = 1
+	// SingleHeader is a B3 encoding that uses a single header named `b3 to
+	// transmit tracing information.
+	SingleHeader B3Encoding = 2
```

Comment:
I don't know what the current style guide says, but for bitmask enums I've often seen the `iota` usage as

```go
const (
    MultiHeader B3Encoding = 1 << iota
    SingleHeader
)
```
In this case, we have only two enum values and unlikely (?) to add any more.

I can understand seeing the auto-incrementing iota and omitted expression replication as over-complication at the risk of simplicity - especially for non Go-idiomatic users.